dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -256,6 +292,322 @@ public ViewAsPageStatus getViewAsStatus(final HttpServletRequest request,
+//            final List<Tag> contentletFoundTags = APILocator.getTagAPI().getTagsByInode(contentlet.getInode());
+//            if (contentletFoundTags != null) {
+//                this.pageFoundTags.addAll(contentletFoundTags);
+//            }
+//        }
+//    }
+
+    private boolean needParseContainerPrefix(final Container container, final String uniqueId,
+            final TemplateLayout templateLayout) {
+        String containerIdOrPath = null;
+
+        if (FileAssetContainerUtil.getInstance().isFileAssetContainer(container)) {
+            containerIdOrPath = FileAssetContainerUtil.getInstance().getFullPath((FileAssetContainer) container);
+        } else {
+            containerIdOrPath = container.getIdentifier();
+        }
+
+        return !ParseContainer.isParserContainerUUID(uniqueId) &&
+                (templateLayout == null || !templateLayout.existsContainer(containerIdOrPath, uniqueId));
+    }
+
+    private Contentlet getContentlet(final PersonalizedContentlet personalizedContentlet,
+            final PageMode mode, final long languageId, final User user) {
+
+        return Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false)?
+                getContentletOrFallback(personalizedContentlet, mode, languageId, user)
+                : getSpecificContentlet(personalizedContentlet, mode, languageId, user);
+    }
+
+    private Contentlet getSpecificContentlet(final PersonalizedContentlet personalizedContentlet,
+            final PageMode mode, final long requestedLanguage, final User user) {
+        try {
+
+            long languageId = this.resolveLanguageId().isPresent() ? this.resolveLanguageId().get()
+                    : requestedLanguage;
+
+            return APILocator.getContentletAPI().findContentletByIdentifier
+                    (personalizedContentlet.getContentletId(), mode.showLive, languageId,
+                            user, mode.respectAnonPerms);
+        } catch (final DotContentletStateException e) {
+            // Expected behavior, DotContentletState Exception is used for flow control
+            return null;
+        } catch (Exception e) {
+            throw new DotStateException(e);
+        }
+    }
+
+    private Contentlet getContentletOrFallback(final PersonalizedContentlet personalizedContentlet,
+            final PageMode mode, final long languageId, final User user) {
+        try {
+
+            final Optional<Contentlet> contentletOpt = APILocator.getContentletAPI().findContentletByIdentifierOrFallback
+                    (personalizedContentlet.getContentletId(), mode.showLive, languageId, user, mode.respectAnonPerms);
+
+            final Contentlet contentlet = contentletOpt.isPresent()
+                    ? contentletOpt.get() : APILocator.getContentletAPI()
+                    .findContentletByIdentifierAnyLanguage(personalizedContentlet.getContentletId());
+
+            return contentlet;
```

Comment:
Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'contentlet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6090726)